src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -127,7 +127,14 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
      */
     public Map<Integer, Float> searchLeaf(LeafReaderContext context, int k) throws IOException {
         final BitSet filterBitSet = getFilteredDocsBitSet(context);
+        final int maxDoc = context.reader().maxDoc();
         int cardinality = filterBitSet.cardinality();
+        /*
+        * If filters match all docs in this segment, then there is no need to do any extra step
+        * and should directly do ANN Search*/
+        if (cardinality ==  maxDoc){
+            return doANNSearch(context, filterBitSet, cardinality, k);
```

Comment:
```suggestion
            return doANNSearch(context, new FixedBitSet(0), 0, k);
```

Also, the code should be run after
```
if (filterWeight != null && cardinality == 0) {
            return Collections.emptyMap();
}
```